second.c

```diff
@@ -0,0 +1,215 @@
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define FILENAME "users.txt"
+
+typedef struct
+{
+    int id;
+    char fname[100];
+    char lname[100];
+    int age;
+} User;
+
+void initializeFile()
+{
+    FILE *file = fopen(FILENAME, "r");
+    if (file == NULL)
+    {
+        file = fopen(FILENAME, "a");
+        printf("File has been created. \n");
+    }
+    fclose(file);
+}
+
+void addUser()
+{
+    User user;
+    int id, found = 0;
+
+    // first opening in read mode to check if the id is unique or not
+    FILE *file = fopen(FILENAME, "r");
+
+    printf("Enter User ID: ");
+    scanf("%d", &id);
+
+    while (fscanf(file, "%d %s %s %d", &user.id, user.fname, user.lname, &user.age) != EOF)
+    {
+        if (user.id == id)
+        {
+            found = 1;
+            printf("User ID already exists. Please enter a new one.\n");
+            break;
+        }
+    }
+
+    if (found)
+    {
+        fclose(file);
+        return;

```

Comment:
having multiple return statement is good practice?

Reply:
Multiple return statements are goods for readability in small function but can complicate longer ones, so I corrected the function for better clarity